redis_collections/lists.py

```diff
@@ -98,16 +98,26 @@ def _get_slice(self, index):
         """Helper for getting a slice."""
         assert isinstance(index, slice)
 
+        # For slices without a step we can use the Redis range function
         def slice_trans(pipe):
-            start, stop = self._recalc_slice(index.start, index.stop)
-            values = pipe.lrange(self.key, start, stop)
-            if index.step:
-                # step implemented by pure Python slicing
-                values = values[::index.step]
+            calc_start, calc_stop = self._recalc_slice(index.start, index.stop)
+            if calc_start == index.stop:
+                return []
+
+            values = pipe.lrange(self.key, calc_start, calc_stop)
             values = [self._unpickle(x) for x in values]
+            pipe.multi()
+            return self._create_new(values, pipe=pipe)
 
+        # Otherwise we'll need to pull the whole list and slice in Python
```

Comment:
I surprise myself with how thoroughly I documented this lib 😄 At that time it was my very first PyPI package.